v2/internal/metrics/arm_client_metrics.go

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package metrics
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+type HttpMethod string
```

Comment:
Is there not already an enum for this defined somewhere else we can reuse?

Reply:
AFAIK, there's nothing in `net/http` and other standard go libs. Have a look at the link below:

https://pkg.go.dev/search?limit=25&m=symbol&q=httpmethod